File: src/main/java/fi/aalto/cs/apluscourses/model/UnexpectedResponseException.java
package fi.aalto.cs.apluscourses.model;

import java.io.IOException;
import org.apache.http.HttpResponse;
import org.jetbrains.annotations.NotNull;

public class UnexpectedResponseException extends IOException {

  private static final long serialVersionUID = -3010286248078758468L;

  @NotNull
  private final transient HttpResponse response;

  /**
   * Constructs a {@link UnexpectedResponseException} with the given response and message.
   *
   * @param response The response to which this exception relates.
   * @param message  The message.
   */
  public UnexpectedResponseException(@NotNull HttpResponse response,
                                     @NotNull String message) {
    super(message);
    this.response = response;
  }

  @NotNull
  public HttpResponse getResponse() {
    return response;
  }
}

File: src/main/kotlin/fi/aalto/cs/apluscourses/model/temp/UnexpectedResponseException.kt
package fi.aalto.cs.apluscourses.model.temp

import org.apache.http.HttpResponse
import java.io.IOException
import java.io.Serial

class UnexpectedResponseException(
    response: HttpResponse,
    message: String
) : IOException(message) {
    @Transient
    private val response: HttpResponse

    /**
     * Constructs a [UnexpectedResponseException] with the given response and message.
     *
     * @param response The response to which this exception relates.
     * @param message  The message.
     */
    init {
        this.response = response
    }

    fun getResponse(): HttpResponse {
        return response
    }

    companion object {
        @Serial
        private val serialVersionUID = -3010286248078758468L
    }
}
